tests/unit_tests/commandline/test_expose.py

```diff
@@ -0,0 +1,72 @@
+"""Unit test asserting functionality for exposing plugin-package commands."""
+
+from importlib import metadata
+import logging
+import pytest
+
+from geoips.errors import PluginPackageNotFoundError
+from geoips.geoips_utils import expose_geoips_commands
+
+LOG = logging.getLogger(__name__)
+
+plugin_packages = [
+    str(ep.value) for ep in metadata.entry_points(group="geoips.plugin_packages")
+]
+
+
+def generate_id(pkg_name):
+    """Generate a test ID for the provided plugin package."""
+    return f"expose {pkg_name}"
+
+
+@pytest.mark.parametrize(
+    "pkg_name", plugin_packages + ["non_existent_pkg"], ids=generate_id
+)
+def test_expose_pkg_cmds(pkg_name, caplog):
+    """Assert various calls to 'expose_geoips_commands' work as expected.
+
+    Parameters
+    ----------
+    pkg_name: str
+        - The name of the plugin package whose command's that will be exposed
+    caplog: pytest caplog fixture
+        - Pytest fixture which captures the log from the corresponding calls
+    """
+    # 35 is the level of LOG.interactive, which we use in expose_geoips_commands
+    caplog.set_level(35)
+    if pkg_name in ["geoips", "data_fusion"]:
+        # Both of these packages have commands. geoips is implemented via poetry, and
+        # data_fusion is implemented via setuptools. Either way, this should work
+        # without an error being raised
+        expose_geoips_commands(pkg_name, LOG)
+        assert f"Available {pkg_name.title()} Commands" in caplog.text
+        eps = list(
+            filter(
+                lambda ep: pkg_name in ep.value,
+                metadata.entry_points().select(group="console_scripts"),
+            )
+        )
+        # Replace calls are needed as we need to filter out the table chars
+        # if the table was split due to terminal size
+        replaced = str(
+            caplog.text.replace("\n", "")
+            .replace(" ", "")
+            .replace(
+                "-",
+                "",
+            )
+            .replace("│", "")
+        )
+        for ep in eps:
+            assert ep.name in replaced  # name of the command
+            assert ep.value in replaced  # path to the command
+    elif pkg_name in plugin_packages:
+        # None of these packaes have commands
+        expose_geoips_commands(pkg_name, LOG)
+        assert f"No '{pkg_name.title()}' Commands were found." in caplog.text
```

Comment:
Do we care that this unit test will automatically break if we add a console script to another repo?  Or if someone installs their own plugin repo that has a console script this unit test will break?

Reply:
Do we want to allow console scripts from other repos? I suppose there's no reason not to. Maybe we can fix this in a follow-on issue?

Reply:
Yes, please fix it.

Reply:
You can resolve this conversation and merge when an issue is open, and linked here!

Reply:
https://github.com/NRLMMD-GEOIPS/geoips/issues/685